e2e/cypress/integration/api/iam/04_projects_api_client_runs_ingestion.ts

```diff
@@ -0,0 +1,291 @@
+      headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+      method: 'POST',
+      url: '/apis/iam/v2beta/apply-rules'
+    });
+
+    waitUntilApplyRulesNotRunning(100);
+  });
+
+  after(() => cy.cleanupV2IAMObjectsByIDPrefixes(cypressPrefix, ['projects', 'policies']));
+
+  it('tagging 7 projects with all the attributes on a client runs node', () => {
+    const nodeId = uuidv4();
+
+    // Ensure there are no nodes matching any of the projects
+    projectsWithRule.forEach(projectWithRule => {
+      cy.request({
+        headers: {
+          'api-token': Cypress.env('ADMIN_TOKEN'),
+          projects: projectWithRule.project.id
+        },
+        method: 'GET',
+        url: '/api/v0/cfgmgmt/nodes?pagination.size=10'
+      }).then((response) => {
+        expect(response.body).to.have.length(0);
+      });
+    });
+
+    // Ingest a node with attribues that match all the projects
+    cy.fixture('converge/avengers1.json').then((node) => {
+      node.organization_name = '75th Rangers';
+      node.chef_server_fqdn = 'example.org';
+      node.node.chef_environment = 'arctic';
+      node.policy_group = 'red_ring';
+      node.policy_name = 'fire';
+      node.node.automatic.roles = ['backend'];
+      node.node.normal.tags = ['v3'];
+      node.entity_uuid = nodeId;
+      cy.request({
+        headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+        method: 'POST',
+        url: '/data-collector/v0',
+        body: node
+      });
+    });
+
+    // wait for the node to be ingested
+    waitForNodes(projectsWithRule[0].project.id, 30);
+
+    // Ensure the node is tagged with the correct project
+    projectsWithRule.forEach(projectWithRule => {
+      cy.request({
+        headers: {
+          'api-token': Cypress.env('ADMIN_TOKEN'),
+          projects: projectWithRule.project.id
+        },
+        method: 'GET',
+        url: '/api/v0/cfgmgmt/nodes?pagination.size=10'
+      }).then((response) => {
+        expect(response.body).to.have.length(1);
+        expect(response.body[0].id).to.equal(nodeId);
+      });
+    });
+
+    // Delete node
+    cy.request({
+      headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+      method: 'POST',
+      url: 'api/v0/ingest/events/chef/node-multiple-deletes',
+      body: {
+        node_ids: [nodeId]
+      },
+      failOnStatusCode: false
+    });
+  });
+});
+
+function waitUntilApplyRulesNotRunning(attempts: number): void {
+  if (attempts === -1) {
+    throw new Error('apply-rules never finished');
+  }
+  cy.request({
+    headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+    url: '/apis/iam/v2beta/apply-rules'
+  }).then((response) => {
+    if (response.body.state === 'not_running') {
+      return;
+    } else {
+      cy.log(`${attempts} attempts remaining: waiting for apply-rules to be not_running`);
+      cy.wait(1000);
+      waitUntilApplyRulesNotRunning(--attempts);
+    }
+  });
+}
+
+function waitForNodes(project: string, maxRetries: number) {
```

Comment:
same here.

might also be worth thinking about how to make this helper usable in the other tests (like you can pass `compliance` and it'll wait for compliance nodes or something)

Reply:
This one might be harder because it will be different for each service. 🤔